service/storage_proxy.cc

```diff
@@ -159,6 +165,26 @@ class storage_proxy::remote {
     netw::connection_drop_slot_t _connection_dropped;
     netw::connection_drop_registration_t _condrop_registration;
 
+    future<std::optional<replica::stale_topology_exception>> try_apply_fence(fencing_token fence,
+        lowres_clock::time_point timeout)
+    {
+        const auto current_version = _sp._shared_token_metadata.get()->get_topology_version();
```

Comment:
We need to use the captured topology, otherwise we have a time-of-check-to-time-of-use race.